### PR TITLE
PLAT-83207 & PLAT-83209: Theme template dynamically setup theme and skin name

### PR DESCRIPTION
**Pull request by JayCanuck**
Updates the outputted source codes files such that they reflect the user-specified theme name.  Additionally supports an optional skin name via `--skin <name>` CLI syntax.

Also updates post-creation user feedback message/information.

Enact-DCO-1.0-Signed-off-by: Jason Robitaille <jason.robitaille@lge.com>

### CODE REVIEW

**Review by sugardave — CHANGES_REQUESTED** [review]
The skin name support is leaving some references to `my-skin`:
- `Skinnable`
- `ThemeDecorator`
- `styles/skin.less`
-  the README

**Comment by JayCanuck** [comment]
Should be good now :+1:

File: packages/theme/index.js [hunk 1]
```diff
@@ -1,13 +1,70 @@
...
+console.log = (log => (data, ...rest) =>
+	typeof data === 'undefined'
+		? log()
+		: typeof data === 'string'
+			? log(data.replace(/Enact app/, 'Enact theme'), ...rest)
+			: log.call(this, data, ...rest))(console.log);
+
 module.exports = {
+	type: 'theme',
+	setup: ({defaultGenerator, directory, name, skin = 'default-skin'}) => {
```
Comment (webOS101) [inline packages/theme/index.js:23]: If we rename 'my-skin' to 'default-skin' then our docs will be wrongly saying to update 'my-skin'.  Should we skip renaming skins if nothing is supplied?  Or, should we also update references in the README.md file that gets copied to the new theme?  How do you set the skin name?  Could we document how to do that if there's a way?

Comment (JayCanuck) [reply packages/theme/index.js:23]: We do update references within the readme.  Conceptually to me, `my-skin` conveys a 'demo-like' purpose, as an example, whereas `default-skin` as the output value in a new project (when an explicit skin name is omitted) fits better and requires no outright change by theme devs (when they only want the singular default skin).

Updating both the template documentation and modifying the in-theme readme for the cli process is a separate ticket Dave is working on concurrently. Usage, however is along the lines: `enact -t theme <theme-name> --skin <skin-name>`

Comment (webOS101) [reply packages/theme/index.js:23]: :nice: Sounds good!

File: packages/theme/package.json [hunk 2]
```diff
@@ -15,5 +15,8 @@
   },
   "publishConfig": {
     "access": "public"
+  },
+  "dependencies": {
```
Comment (Djspaceg) [inline packages/theme/package.json:19]: Should this be a devDependency instead?

Comment (JayCanuck) [reply packages/theme/package.json:19]: Nah, `chalk` is used by the template at `enact create` time; it's needed for operation of the template and is needed to be included in the package tree itself. It won't be included in the output theme just needed for the template itself.

**Review by Djspaceg — APPROVED** [review]

File: packages/theme/index.js [hunk 3]
```diff
@@ -1,13 +1,70 @@
...
+const capEachWord = text => text.split(/[-_\s]/).map(capitalize).join('');
+const ilibVar = name => `ILIB_${name.toUpperCase().replace(/[-_\s]/g, '_')}_PATH`;
+
+const sourceFiles = (dir, action) => {
+	const list = fs.readdirSync(dir);
+	list.forEach(item => {
+		const itemPath = path.join(dir, item);
+		if (fs.statSync(itemPath).isDirectory()) {
+			sourceFiles(itemPath, action);
+		} else if (/\.(css|less|js|jsx|md)$/.test(item)) {
```
Comment (webOS101) [inline packages/theme/index.js]: Should we update this to include `ts|tsx` for future-proofing?

File: packages/theme/index.js [hunk 4]
```diff
@@ -1,13 +1,70 @@
...
+			action(itemPath);
+		}
+	})
+};
+
+// Temporary fallback until ENYO-6133 is merged
```
Comment (webOS101) [inline packages/theme/index.js]: We can remove this fallback, I think.  I just merged ENYO-6133.

File: packages/theme/index.js [hunk 5]
```diff
@@ -20,6 +77,18 @@ module.exports = {
 		console.log(chalk.cyan('	npm run lint'));
 		console.log('		Lints the source code.');
 		console.log();
+		console.log('We suggest that you begin by typing:');
+		if (path.resolve(process.cwd()) !== path.resolve(directory)) {
+			console.log(chalk.cyan('	cd ' + path.relative(process.cwd(), directory)));
+		}
+		console.log('	' + chalk.cyan('npm run transpile'));
+		console.log('	' + chalk.cyan('npm link build'));
+		console.log('This will build your theme library into ES5/CSS and then link that '
```
Comment (webOS101) [inline packages/theme/index.js:78]: Normally, for development internally, we link to the 'raw' version of the template for debugging.  Why are we suggesting using the transpiled one here?

**Review by webOS101 — CHANGES_REQUESTED** [review]

File: packages/theme/index.js (hunk 5 above)
Comment (JayCanuck) [reply packages/theme/index.js:78]: The ability to link "raw" templates is something solely affording to "@enact/*" libraries. We have special handling for `node_modules/@enact/*` to have webpack treat our code there like app-level code  (typescript/react/babel transpiling).  This is not the community standard and was done solely as an internal team "special case".  So for the case of 3rd party themes, yep, "transpile -> link `./build`" is the correct workflow.

File: packages/theme/index.js (hunk 3 above)
Comment (JayCanuck) [reply packages/theme/index.js]: Good idea, will do

File: packages/theme/index.js (hunk 4 above)
Comment (JayCanuck) [reply packages/theme/index.js]: Can do :+1:

File: packages/theme/index.js (hunk 5 above)
Comment (webOS101) [reply packages/theme/index.js:78]: Aha, of course.

**Review by webOS101 — APPROVED** [review]
:awesome:

**Pull request merged**